api/ws-server.js

```diff
@@ -0,0 +1,49 @@
+const WebSocket = require('ws')
+
+const logger = require('./config/winston')
+const { appEmitter } = require('./emitters.js')
```

Comment:
This seems to be functioning as a communication bus. I would use a different variable name here `appEmitterBus` perhaps. Also since this seems to be a global bus, where event names are global variables, I would avoid event names such as "solution" and use something more specific such as "solution-judgement-available".